packages/backend/src/models/OAuth2Model.ts

```diff
@@ -0,0 +1,300 @@
+                'oauth2_clients.client_id',
+                'oauth2_clients.client_secret',
+                'oauth2_clients.redirect_uris',
+                'oauth2_clients.grants',
+                'oauth2_clients.scopes',
+                'oauth2_clients.client_name',
+                'users.user_uuid',
+            )
+            .leftJoin(
+                'oauth2_clients',
+                'oauth2_access_tokens.client_id',
+                'oauth2_clients.client_id',
+            )
+            .leftJoin('users', 'oauth2_access_tokens.user_id', 'users.user_id')
+            .where('oauth2_access_tokens.access_token', accessToken)
+            .first();
+
+        if (!result) {
+            return false;
+        }
+
+        return {
+            accessToken: result.access_token,
+            accessTokenExpiresAt: new Date(result.expires_at),
+            scope: result.scope,
+            client: {
+                id: result.client_id,
+                redirectUris: result.redirect_uris,
+                grants: result.grants,
+            },
+            user: {
+                userId: result.user_id,
+                userUuid: result.user_uuid,
+                organizationUuid: result.organization_uuid,
+            },
+        };
+    }
+
+    async revokeToken(token: Token): Promise<boolean> {
+        if (!token.refreshToken) {
+            return false;
+        }
+
+        const result = await this.database('oauth2_refresh_tokens')
+            .where('refresh_token', token.refreshToken)
+            .del();
+
+        return result > 0;
+    }
+
+    async getRefreshToken(refreshToken: string): Promise<Token | false> {
+        const result = await this.database('oauth2_refresh_tokens')
+            .select(
+                'oauth2_refresh_tokens.*',
+                'oauth2_clients.client_id',
+                'oauth2_clients.client_secret',
+                'oauth2_clients.redirect_uris',
+                'oauth2_clients.grants',
+                'oauth2_clients.scopes',
+                'oauth2_clients.client_name',
+            )
+            .leftJoin(
+                'oauth2_clients',
+                'oauth2_refresh_tokens.client_id',
+                'oauth2_clients.client_id',
+            )
+            .where('oauth2_refresh_tokens.refresh_token', refreshToken)
+            .first();
+
+        if (!result) {
+            return false;
+        }
+
+        return {
+            accessToken: '',
+            refreshToken: result.refresh_token,
+            refreshTokenExpiresAt: new Date(result.expires_at),
+            scope: result.scope,
+            client: {
+                id: result.client_id,
+                redirectUris: result.redirect_uris,
+                grants: result.grants,
+            },
+            user: {
+                userId: result.user_id,
+                organizationUuid: result.organization_uuid,
+            },
+        };
+    }
+
+    async generateAccessToken(
```

Comment:
Is this where we could conditionally generate a personal access token instead of something else depending on the scopes provided/